engine/contact/contact_test.go

```diff
@@ -0,0 +1,150 @@
+package contact_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/engine/contact/testabilities"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/tester/fixtures"
+)
+
+func Test_ClientService_AdminCreateContact_Success(t *testing.T) {
+	tests := map[string]struct {
+		contactPaymail   string
+		creatorPaymail   string
+		fullName         string
+		metadata         *engine.Metadata
+		expectedError    error
+		expectedStatus   engine.ContactStatus
+		expectedFullName string
+	}{
+		"Happy path without metadata": {
+			contactPaymail:   fixtures.RecipientExternal.DefaultPaymail(),
+			creatorPaymail:   fixtures.Sender.DefaultPaymail(),
+			fullName:         "John Doe",
+			metadata:         nil,
+			expectedError:    nil,
+			expectedStatus:   engine.ContactNotConfirmed,
+			expectedFullName: "John Doe",
+		},
+		"Happy path with metadata": {
+			contactPaymail: fixtures.RecipientExternal.DefaultPaymail(),
+			creatorPaymail: fixtures.Sender.DefaultPaymail(),
+			fullName:       "John Doe",
+			metadata: &engine.Metadata{
+				"key1": "value1",
+				"key2": 420,
+			},
+			expectedError:    nil,
+			expectedStatus:   engine.ContactNotConfirmed,
+			expectedFullName: "John Doe",
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			//given:
+			given, then := testabilities.New(t)
+
+			service, cleanup := given.Engine()
+			defer cleanup()
+
+			//when:
+			res, err := service.AdminCreateContact(context.Background(), tt.contactPaymail, tt.creatorPaymail, tt.fullName, tt.metadata)
+
+			//then:
+			then.NoError(err).WithResponse(res).WithStatus(tt.expectedStatus).WithFullName(tt.expectedFullName)
+		})
+	}
+}
+
+func Test_ClientService_AdminCreateContact_PKIRetrievalFail(t *testing.T) {
+	t.Run("Should fail with PKI retrieval", func(t *testing.T) {
+		//given:
+		given, then := testabilities.New(t)
+
+		service, cleanup := given.Engine()
+		defer cleanup()
+
+		//and:
+		given.PaymailClient().WillRespondOnCapability(paymail.BRFCPki).WithInternalServerError()
+
+		//when:
+		res, err := service.AdminCreateContact(context.Background(), fixtures.RecipientExternal.DefaultPaymail(), fixtures.Sender.DefaultPaymail(), "John Doe", nil)
+
+		//then:
+		then.ErrorIs(err, spverrors.ErrGettingPKIFailed).WithNilResponse(res)
```

Comment:
If you would go with structure more like this:
```suggestion
		then.ContactCreated(res).WithError(err).ThatIs(spverrors.ErrGettingPKIFailed)
```
then in `WithError` you could require `res` (contact) to be `nil`, as this is what we always expect when there is an error.

ℹ️ You can find example of something like that in [create_draft_test.go](https://github.com/bitcoin-sv/spv-wallet/blob/3312ad9f92c0acb11723db02e18850535b08b549/engine/transaction/outlines/create_draft_test.go#L51)

Reply:
done 🚀